DataFormats/FTLRecHit/src/FTLUncalibratedRecHit.cc

```diff
@@ -0,0 +1,105 @@
+#include "DataFormats/FTLRecHit/interface/FTLUncalibratedRecHit.h"
+#include <math.h>
+
+FTLUncalibratedRecHit::FTLUncalibratedRecHit() :
+  amplitude_(0.), pedestal_(0.), jitter_(0.), chi2_(10000.), OOTamplitude_(0.), OOTchi2_(10000.), flags_(0), aux_(0) { }
+
+FTLUncalibratedRecHit::FTLUncalibratedRecHit(const DetId& id, float ampl, float ped,
+					     float jit, float chi2, uint32_t flags, uint32_t aux) :
+  amplitude_(ampl), pedestal_(ped), jitter_(jit), chi2_(chi2), OOTamplitude_(0.), OOTchi2_(10000.), flags_(flags), aux_(aux), id_(id) { }
+
+FTLUncalibratedRecHit::~FTLUncalibratedRecHit() {
+}
+
+bool FTLUncalibratedRecHit::isSaturated() const {
+  return FTLUncalibratedRecHit::checkFlag(kSaturated);
+}
+
+
+float FTLUncalibratedRecHit::jitterError() const {
+  // stored in ps, but return BXs to match with jitter units
+  uint32_t jitterErrorBits = 0xFF & aux_;
+  // all bits off --> time reco bailed out (return negative value)
+  if( (0xFF & jitterErrorBits) == 0x00)
+    return -1;
+  // all bits on  --> time error over 5 ns (return large value)
+  if( (0xFF & jitterErrorBits) == 0xFF)
+    return 10000;
+
+  float LSB = 1.26008;
+  uint8_t exponent = jitterErrorBits>>5;
+  uint8_t significand = jitterErrorBits & ~(0x7<<5);
+  return (float)(pow(2.,exponent)*significand*LSB)/(25.*1000);
+}
+
+void FTLUncalibratedRecHit::setJitterError( float jitterErr ) {
+  // use 8 bits (3 exp, 5 mant) and store in ps
+  // has range of 5 ps - 5000 ps
+  // expect input in BX units
+  // all bits off --> time reco bailed out
+  if(jitterErr < 0) {
+    aux_ = (~0xFF & aux_);
+    return;
+  }
+  // all bits on  --> time error over 5 ns
+  if(25*jitterErr >= 5) {
+    aux_ = (0xFF | aux_);
+    return;
+  }
+
+  float LSB = 1.26008;
+  float quantityInLSB = (1000*25*jitterErr)/LSB;
+  int log2OfQuantity = (int) (log2( quantityInLSB ));
+  int exponentTmp = log2OfQuantity - 4;
+  uint8_t exponent=0;
+  if (exponentTmp>0) exponent = exponentTmp;
+  uint8_t significand = (int) ( lround( quantityInLSB / pow(2.,exponent) )   );
```

Comment:
Ditto to comment above. Not used yet because of state of reconstruction and will be rewritten before it is used.